src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2092,6 +2031,168 @@ WorkerExtendedOpNode(MultiExtendedOp *originalOpNode,
 }
 
 
+/*
+ * ProcessWorkerExpressionList is a utility function which processes the
+ * expressions that are intended to be added to the worker target list.
+ *
+ * In summary, the function gets a list of expressions, converts them to target
+ * entries and updates all the necessary fields such that the expression is correctly
+ * added to the worker query's target list.
+ *
+ * Adding an expression to the target list includes generating a target entry,
+ * appending it to the newTargetEntryList and incrementing the projection number.
+ *
+ * If the expression is requested to be added to the group clause, the entry
+ * is appended to groupClauseList and sort group reference is incremented.
+ */
+static void
+ProcessWorkerExpressionList(List *expressionList, TargetEntry *originalTargetEntry,
+							bool addToGroupByClause, List **newTargetEntryList,
+							AttrNumber *targetProjectionNumber, List **groupClauseList,
+							Index *nextSortGroupRefIndex)
+{
+	ListCell *newExpressionCell = NULL;
+
+	/* now create target entries for each new expression */
+	foreach(newExpressionCell, expressionList)
+	{
+		Expr *newExpression = (Expr *) lfirst(newExpressionCell);
+		TargetEntry *newTargetEntry = NULL;
+
+		/* generate and add the new target entry to the target list */
+		newTargetEntry =
+			GenerateWorkerTargetEntry(originalTargetEntry, newExpression,
+									  *targetProjectionNumber);
+		(*targetProjectionNumber)++;
+		(*newTargetEntryList) = lappend(*newTargetEntryList, newTargetEntry);
```

Comment:
Since pointer dereferencing adds some mental overhead, maybe turn these into variables and set the pointers at the end?

Reply:
I actually have a slight preference to keep them dereferenced pointers and create that cognitive overhead. On the high level, one of the goals of these series of PRs is to create a clear distinction between the inputs and the outputs. And, it feels like these dereferened pointers helps that.

That said if you think its too disturbing, I'm happy to change them as well.